panel/panel.sublime-syntax

```diff
@@ -37,13 +44,21 @@ contexts:
     - match: (?=$)
       pop: true
 
+  ensure-warning-other-scope:
+    - meta_scope: meta.linter.body.sublime_linter
+    - match: (?=$)
+      pop: true
+
   expect-linter-severity:
     - include: pop-at-end
     - match: \berror\b
-      scope: markup.error.sublime_linter
+      scope: meta.linter.error-type.sublime_linter markup.error.sublime_linter
```

Comment:
the meta is actually usually intended to indicate you're matching a group of other, more specific scopes: https://www.sublimetext.com/docs/3/scope_naming.html#meta

> Meta scopes are used to scope larger sections of code or markup, generally containing multiple, more specific scopes.

Reply:
Suggestion? 

Usually you would go for `x.linter.problem.error`, `x.linter.problem.warning`, and for other types just `x.linter.problem`. 

I really don't know why we use `entity.type.linter-name` but `storage.type.error-code`. Why do we change the namespace from _entity_ to _storage_?

Reply:
> Why do we change the namespace from entity to storage?

To differentiate them in a way that will result in a different color in most color schemes. *No* color schemes, now or in the future, will have specified something for `linter-name`. So if they're both just `entity.type` both will look exactly the same. There should be something to differentiate them in a way that makes sense semantically, and there should be something in there that helps existing color schemes assign colors.

Reply:
I could go with `entity.name.class` for linter-name, but the default color schemes don't have anything for `entity.name.label` so the code would look just like the name. Which isn't bad, but I like some differentiation there. `entity.name.function` would work though. `entity.name.tag` would work for the error types, but we already have `markup.error/warning` there ensuring that we get signal colors by default.

Reply:
That's not what I'm seeing. 

![image](https://user-images.githubusercontent.com/8558/44345051-79c60080-a492-11e8-81bd-a90f4e26ea8a.png)

Most default schemes actually **don't** have anything for `*.class`, but for `label` and for `tag`. (Basically except 'Celeste')

**None** of the default schemes define `markup.error` or `markup.warning`. Here I wanted the scope to be `entity.name.tag markup.error`, in case someone has that defined.

Now, I think for our purpose class, label, tag can be used interchangeably. So, we have at most 3 different default colors, we can choose from. We can use all three or make it less Kindergarten. E.g. `entity.name.label.error-type`, `entity.name.label.linter-name` and `entity.name.label.error-code`

Reply:
> None of the default schemes define markup.error or markup.warning

Dang, you're right!? What crap editor is this anyway?

Ok, so I won't really get the control that I want. Let's keep it simple then.

We do have the colorish scopes to get the signal colors for error/warning like we do elsewhere. Let's use the same scopes there as our defaults: `region.yellowish markup.warning.sublime_linter` and `region.redish markup.error.sublime_linter`. I don't believe we need an additional `entity.name` scope there because there are already plenty of hooks to customize it if you want. 

For the others let's use `entity.name.label.linter-name` and `entity.name.label.error-code`. I'd like to have the differentiation, but we don't have to color all the things.

Reply:
Oh, right, we can't use the colorish scopes because they also do the background color. Dang.

Reply:
Ok, you've got the overall idea and I'm apparently lacking the sharpness to really add anything of use here. Final thing I'd like to say is that I think the error-type should have a really different scope than the linter:error block so it doesn't all look the same in all the default color schemes. Beyond that I make my own color schemes anyway.